components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONRequestParser.java

```diff
@@ -105,113 +105,124 @@ public static RequestCtx parse(String jsonRequest) throws JsonParseException, Re
-                        if (jsonCategory.has(EntitlementEndpointConstants.CATEGORY_ID)) {
-                            category = stringCateogryToURI(jsonCategory
-                                    .get(EntitlementEndpointConstants.CATEGORY_ID)
-                                    .getAsString());
-                        }
-                    case EntitlementEndpointConstants.CATEGORY_RESOURCE:
-                    case EntitlementEndpointConstants.CATEGORY_ACTION:
-                    case EntitlementEndpointConstants.CATEGORY_ENVIRONMENT:
-                    case EntitlementEndpointConstants.CATEGORY_ACCESS_SUBJECT:
-                    case EntitlementEndpointConstants.CATEGORY_RECIPIENT_SUBJECT:
-                    case EntitlementEndpointConstants.CATEGORY_INTERMEDIARY_SUBJECT:
-                    case EntitlementEndpointConstants.CATEGORY_CODEBASE:
-                    case EntitlementEndpointConstants.CATEGORY_REQUESTING_MACHINE:
-
-                        if (category == null) {
-                            category = stringCateogryToURI(jsonAttribute.getKey());
-                        }
-                        if (jsonCategory.has(EntitlementEndpointConstants.ID)) {
-                            id = jsonCategory.get(EntitlementEndpointConstants.ID).getAsString();
-                        }
-                        if (jsonCategory.has(EntitlementEndpointConstants.CONTENT)) {
-                            ByteArrayInputStream inputStream;
-                            DocumentBuilderFactory dbf;
-                            Document doc = null;
-
-                            String xmlContent = stringContentToXMLContent(jsonCategory
-                                    .get(EntitlementEndpointConstants.CONTENT)
-                                    .getAsString());
-                            inputStream = new ByteArrayInputStream(xmlContent.getBytes());
-                            dbf = IdentityUtil.getSecuredDocumentBuilderFactory();
-                            dbf.setNamespaceAware(true);
-
-                            try {
-                                doc = dbf.newDocumentBuilder().parse(inputStream);
-                            } catch (Exception e) {
-                                throw new JsonParseException("DOM of request element can not be created from String");
-                            } finally {
-                                try {
-                                    inputStream.close();
-                                } catch (IOException e) {
-                                    throw new JsonParseException("DOM of request element can not be created from String");
-                                }
-                            }
-                            if (doc != null) {
-                                content = doc;
-                            }
-                        }
+        RequestCtx requestCtx = new RequestCtx(null, categories, returnPolicyIdList, combinedDecision,
```

Comment:
Is there any specific reason to have a variable **requestCtx**. Can't you straightly return the new RequestCtx object?